pkg/scheduler/ugm/manager.go

```diff
@@ -365,127 +381,159 @@ func (m *Manager) internalProcessConfig(cur configs.QueueConfig, queuePath strin
 }
 
-func (m *Manager) processUserConfig(user string, limitConfig *LimitConfig, queuePath string, userLimits map[string]bool) error {
-	if err := m.setUserLimits(user, limitConfig, queuePath); err != nil {
-		return err
-	}
-	userLimits[user] = true
-	return nil
-}
+// clearEarlierSetLimits Clear already configured limits of users and groups for which limits have been configured before but not now
+func (m *Manager) clearEarlierSetLimits(newUserLimits map[string]map[string]*LimitConfig, newGroupLimits map[string]map[string]*LimitConfig) {
+	// Clear already configured limits of group for which limits have been configured before but not now
+	m.clearEarlierSetGroupLimits(newGroupLimits)
 
-func (m *Manager) processGroupConfig(group string, limitConfig *LimitConfig, queuePath string, groupLimits map[string]bool) error {
-	if err := m.setGroupLimits(group, limitConfig, queuePath); err != nil {
-		return err
-	}
-	groupLimits[group] = true
-	return nil
+	// Clear already configured limits of user for which limits have been configured before but not now
+	m.clearEarlierSetUserLimits(newUserLimits)
 }
 
-// clearEarlierSetLimits Clear already configured limits of users and groups for which limits have been configured before but not now
-func (m *Manager) clearEarlierSetLimits(userLimits map[string]bool, groupLimits map[string]bool, queuePath string) error {
-	// Clear already configured limits of group for which limits have been configured before but not now
-	for _, gt := range m.groupTrackers {
-		appUsersMap := m.clearEarlierSetGroupLimits(gt, queuePath, groupLimits)
-		if len(appUsersMap) > 0 {
-			for app, user := range appUsersMap {
-				ut := m.userTrackers[user]
-				ut.setGroupForApp(app, nil)
+func (m *Manager) clearEarlierSetUserLimits(newUserLimits map[string]map[string]*LimitConfig) {
+	if len(m.userLimits) == 0 {
+		return
+	}
+	m.RLock()
+	defer m.RUnlock()
+	// Need to clear user config only when
+	// 1. config set earlier but not now
+	// 2. user already tracked for the queue path
+	for queuePath, limitConfig := range m.userLimits {
+		// Is queue path exists?
+		if newUserLimit, ok := newUserLimits[queuePath]; !ok {
+			for u := range limitConfig {
+				if ut, utExists := m.userTrackers[u]; utExists {
+					m.clearEarlierSetUserLimit(ut, queuePath)
+				}
+			}
+		} else {
+			// Queue path exists. Is user exists?
+			for u := range limitConfig {
+				if _, ulExists := newUserLimit[u]; !ulExists {
+					if ut, utExists := m.userTrackers[u]; utExists {
+						m.clearEarlierSetUserLimit(ut, queuePath)
+					}
+				}
 			}
 		}
 	}
-
-	// Clear already configured limits of user for which limits have been configured before but not now
-	for _, ut := range m.userTrackers {
-		m.clearEarlierSetUserLimits(ut, queuePath, userLimits)
-	}
-	return nil
 }
 
-func (m *Manager) clearEarlierSetUserLimits(ut *UserTracker, queuePath string, userLimits map[string]bool) {
+func (m *Manager) clearEarlierSetUserLimit(ut *UserTracker, queuePath string) {
```

Comment:
This really removes, while the previous function does the looping:
the names look too much alike difficult to understand the difference between the two.